python/run-tests.py

```diff
@@ -77,13 +79,33 @@ def run_individual_python_test(test_name, pyspark_python):
         'PYSPARK_PYTHON': which(pyspark_python),
         'PYSPARK_DRIVER_PYTHON': which(pyspark_python)
     })
+
+    # Create a unique temp directory under 'target/' for each run. The TMPDIR variable is
+    # recognized by the tempfile module to override the default system temp directory.
+    target_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), 'target'))
+    if not os.path.isdir(target_dir):
+        os.mkdir(target_dir)
+    tmp_dir = os.path.join(target_dir, str(uuid.uuid4()))
+    if not os.path.isdir(tmp_dir):
```

Comment:
shouldn't it fail or create a new uuid if the directory is already there?

Reply:
Unlikely to happen but sure.